examples/hosts/app-integration/external-controller/src/FocusTracker.ts

```diff
@@ -0,0 +1,134 @@
+     */
+    private readonly focusMap = new Map<string, boolean>();
+
+    private readonly focusSignalFn = (message: IInboundSignalMessage) => {
+        const userId: string = message.content.userId;
+        const hasFocus: boolean = message.content.focus;
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const clientId: string = message.clientId!;
+        this.focusMap.set(`${userId} ${clientId}`, hasFocus);
+        this.emit("focusChanged");
+    };
+
+    private _audience: IServiceAudience<IMember> | undefined;
+    public get audience(): IServiceAudience<IMember> {
+        if (this._audience === undefined) {
+            throw new Error("no audience");
+        }
+        return this._audience;
+    }
+    public set audience(newAudience: IServiceAudience<IMember>) {
+        if (this._audience !== undefined) {
+            throw new Error("set audience only once");
+        }
+        this._audience = newAudience;
+        this._audience.on("membersChanged", () => {
+            this.pruneFocusMap();
+            this.emit("focusChanged");
+        });
+    }
+
+    public static get Name() { return "@fluid-example/focus-tracker"; }
+
+    public static readonly factory = new DataObjectFactory<FocusTracker, undefined, undefined, IFocusTrackerEvents>
+    (
+        FocusTracker.Name,
+        FocusTracker,
+        [],
+        {},
+    );
+
+    protected async hasInitialized() {
+        window.addEventListener("focus", () => {
+            this.sendFocusSignal(true);
+        });
+        window.addEventListener("blur", () => {
+            this.sendFocusSignal(false);
+        });
+        this.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
+            if (message.type === FocusTracker.focusSignalType &&
+                this.runtime.connected &&
+                // eslint-disable-next-line no-null/no-null
+                message.clientId !== null) {
+                this.focusSignalFn(message);
+            }
+        });
+        this.runtime.on("connected", () => {
+            this.sendFocusSignal(document.hasFocus());
+        });
+    }
+
+    /**
+     * Alert all connected clients that there has been a change to a client's focus
+     */
+    private sendFocusSignal(hasFocus: boolean) {
+        this.runtime.submitSignal(
+            FocusTracker.focusSignalType,
+            { userId: this.audience.getMyself()?.userId, focus: hasFocus },
+        );
+    }
+
+    /**
+     * Go through the current audience to remove entries in our focus map on clients
+     * that are no longer present
+     */
+    private pruneFocusMap() {
+        this.focusMap.forEach((value, key) => {
+            const [, clientId] = key.split(" ");
+            if (this.audience.getMemberByClientId(clientId) === undefined) {
+                this.focusMap.delete(key);
+            }
+        });
+    }
+
+    public getPresences(): string[] {
```

Comment:
Instead of an array of statuses here, could we consider returning a structure that gives a Map of with key `userId` and value another Map?
And on the second Map, key is connectionId and the value is an enum with the status?
Then, someone looking at this example could use the FocusTracker  DataObject in a completely different app where they can make a decision on their string content based off the status enum